pkg/kubectl/cmd/util/shortcut_restmapper.go

```diff
@@ -112,6 +112,8 @@ var UserResources = []schema.GroupResource{
 	{Group: "batch", Resource: "jobs"},
 	{Group: "extensions", Resource: "deployments"},
 	{Group: "extensions", Resource: "replicasets"},
+	{Group: "extensions", Resource: "ingresses"},
+	{Group: "extensions", Resource: "daemonsets"},
```

Comment:
@liggitt Can you please show us more detail for why do we such rules as https://github.com/kubernetes/community/blob/master/contributors/devel/kubectl-conventions.md#rules-for-extending-special-resource-alias---all ?

Perhaps adding some detail for `why` maybe better in the document.